fix: `build.modulePreload.resolveDependencies` is optimizable (#16083)
vitejs/vite: docs/config/build-options.md

  url: string,
  deps: string[],
  context: {
    hostId: string
    hostType: 'html' | 'js'
  },
) => string[]
```

> The `resolveDependencies` function will be called for each dynamic import with a list of the chunks it depends on, and it will also be called for each chunk imported in entry HTML files. A new dependencies array can be returned with these filtered or more dependencies injected, and their paths modified. The `deps` paths are relative to the `build.outDir`. Returning a relative path to the `build.outDir` is allowed.

Author: ```suggestion
The `resolveDependencies` function will be called for each dynamic import with a list of the chunks it depends on, and it will also be called for each chunk imported in entry HTML files. A new dependencies array can be returned with these filtered or more dependencies injected, and their paths modified. The `deps` paths are relative to the `build.outDir`. The return value should be a relative path to the `build.outDir`.
```
It doesn't allow returning an absolue path because `toOutputFilePathWithoutRuntime ` and `toOutputFilePathInJS` doesn't support that.
